### ISSUE TITLE
`vec_box` doesn't take scopes into account

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust-clippy#6236]
I tried this code:

```rust
mod inner {
    pub struct S;
}

mod inner2 {
    use crate::inner::S;

    pub fn f() -> Vec<Box<S>> {
        vec![]
    }
}
```

I expected to see this happen: Clippy suggests turning `Vec<Box<S>>` into `Vec<S>`

Instead, this happened: Clippy suggests turning it into `Vec<inner::S>`.

```
error[E0433]: failed to resolve: use of undeclared crate or module `inner`
  --> src/lib.rs:10:23
   |
10 |     pub fn f() -> Vec<inner::S> {
   |                       ^^^^^ use of undeclared crate or module `inner`
Original diagnostics will follow.

warning: `Vec<T>` is already on the heap, the boxing is unnecessary.
  --> src/lib.rs:10:19
   |
10 |     pub fn f() -> Vec<Box<S>> {
   |                   ^^^^^^^^^^^ help: try: `Vec<inner::S>`
   |
   = note: `#[warn(clippy::vec_box)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#vec_box
```

### Meta

- `cargo clippy -V`: clippy 0.0.212 (ffa2e7a 2020-10-24)
- `rustc -Vv`:
```
rustc 1.47.0 (18bf6b4f0 2020-10-07)
binary: rustc
commit-hash: 18bf6b4f01a6feaf7259ba7cdae58031af1b7b39
commit-date: 2020-10-07
host: x86_64-unknown-linux-gnu
release: 1.47.0
LLVM version: 11.0
```

@rustbot modify labels: +A-suggestion +L-suggestion-causes-error
